app/javascript/components/patient/follow_up_flag/FollowUpFlag.js

```diff
@@ -0,0 +1,278 @@
+          .then(() => {
+            location.href = window.BASE_PATH;
+          })
+          .catch(error => {
+            reportError(error);
+            this.setState({ loading: false });
+          });
+      } else {
+        axios
+          .post(window.BASE_PATH + '/patients/' + this.props.patient.id + '/follow_up_flag', {
+            follow_up_reason: this.state.follow_up_reason,
+            follow_up_note: this.state.follow_up_note,
+            clear_flag: this.state.clear_flag,
+            clear_flag_reason: this.state.clear_flag_reason,
+            apply_to_household: this.state.apply_to_household,
+            apply_to_household_ids: this.state.apply_to_household_ids,
+          })
+          .then(() => {
+            // Reload the page to see the flag on the monitoree's page
+            location.reload();
+          })
+          .catch(err => {
+            reportError(err?.response?.data?.error ? err.response.data.error : err, false);
+          });
+      }
+    });
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <Modal.Body className="modal-follow-up-flag-body">
+          <Form.Group className="flag-radio-buttons">
+            <Form.Check
+              type="radio"
+              name="flag_for_follow_up_option"
+              className="pr-50"
+              id="set_flag_for_follow_up"
+              label="Set Follow-up Flag"
+              onChange={this.handleChange}
+              checked={!this.state.clear_flag}
+            />
+            <Form.Check
+              type="radio"
+              name="flag_for_follow_up_option"
+              id="clear_flag_for_follow_up"
+              label="Clear Follow-up Flag"
+              disabled={this.state.clear_flag_disabled}
+              onChange={this.handleChange}
+              checked={this.state.clear_flag}
+            />
+          </Form.Group>
+          {!this.state.clear_flag && (
+            <Form.Group>
+              <Form.Label>
+                Please select a reason for being flagged for follow-up. If a monitoree is already flagged, this reason will replace any previously selected
+                reason.
+              </Form.Label>
+              <Form.Control
+                as="select"
+                size="lg"
+                className="form-square mb-25"
+                id="follow_up_reason"
+                value={this.state.follow_up_reason}
+                onChange={this.handleChange}>
+                <option></option>
+                <option>Deceased</option>
+                <option>Duplicate</option>
+                <option>High-Risk</option>
+                <option>Hospitalized</option>
+                <option>In Need of Follow-up</option>
+                <option>Lost to Follow-up</option>
+                <option>Needs Interpretation</option>
+                <option>Quality Assurance</option>
+                <option>Other</option>
+              </Form.Control>
+              <Form.Group>
```

Comment:
for the same reasoning here, this `Form.Group` needs `controlId="follow_up_note"` and then you can remove the id from the control